h/static/scripts/directive/group-list.js

```diff
@@ -19,13 +21,27 @@ function GroupListController($scope, $window) {
   };
 
   $scope.leaveGroup = function (groupId) {
-    var groupName = $scope.groups.get(groupId).name;
+    var groupName = groups.get(groupId).name;
     var message = 'Are you sure you want to leave the group "' +
       groupName + '"?';
     if ($window.confirm(message)) {
-      $scope.groups.leave(groupId);
+      groups.leave(groupId);
     }
   }
+
+  $scope.focusGroup = function (groupId) {
+    groups.focus(groupId);
+  }
+
+  $scope.allGroups = groups.all();
+  $scope.$on(events.GROUPS_CHANGED, function () {
+    $scope.allGroups = groups.all();
```

Comment:
Ah, wait, I see. It is necessary because a digest cycle won't automatically be triggered when the session data updates.